src/freenas/usr/local/sbin/firmware_update.py

```diff
@@ -21,9 +22,9 @@
 FAILED_UPDATE_SENTINEL = '/data/.hba_firmware_update_fail'
 UPDATE_SENTINEL = '/data/.hba_firmware_update'
 
-SAS92_VERSION = 20
-SAS93_VERSION = 16
-HBA94_VERSION = 15
+SAS92_VERSION = version.parse("20.0.0.0")
```

Comment:
@darkfiberiru will you be able to help me put the current firmware versions here?

Reply:
Sorry github notifications are  a mess for me and have been sick. Put current firmware how? Like how to get it or how list of them or???

Reply:
It will not work like this, since we have different minor versions for different board models.  It has to be encoded in file name probably.  For 92xx and 93xx the value can be easily seen just as a string in the binary.  For 94xx something has changed.

Reply:
Are there multiple firmwares in each file?

Reply:
Each card has a individual firmware that is fixed. With each family of cards being same major version(atleast so far). We have one file per card.... issue is we previously didn't do a good job of document what that version is. According to mav all but 9400 cards you can find string in binary firmware file.

Reply:
https://github.com/freenas/firmware